### PR TITLE
Feature/mx-1533 graph id provider

### PR DESCRIPTION
**Pull request by cutoffthetop**
# PR Context
this change allows us to stop re-creating models from mex-common in mex-backend just to sidestep the [set_identifiers](https://github.com/robert-koch-institut/mex-common/blob/0.19.1/mex/common/models/extracted_data.py#L73) validator. now we have a graph-identity-provider that we can use in the backend to assign and fetch identities directly from the database.

# Added
- graph identity provider that assigns ids to extracted data
- generalize type enums into DynamicStrEnum superclass
- seed primary source for mex on connector init
- test fixture that makes Identifiers deterministic

# Changes
- assign name to uniqueness constraint
- use graph identity provider in identity endpoints
- add module name to dynamic models for better debugging
- allow 'MergedThing' as well as 'Thing' as entityType query parameter

# Removed
- remove dynamic extracted model classes and use those from mex-common

# Fixed
- don't allow identifierInPrimarySource changes on node updates

### CODE REVIEW

**Review by mr-kamran-ali — COMMENTED** [review]
All tests pass, coudnt find anything major. Just one small question.

File: .github/workflows/cookiecutter.yml [hunk 1]
```diff
@@ -2,7 +2,9 @@ name: Cookiecutter
 
 on:
   push:
-    branches: ["main"]
+  pull_request:
+  schedule:
+    - cron: '0 0 * * *'
```
[the remark this answers is not in the document]
Comment (cutoffthetop) [reply .github/workflows/cookiecutter.yml:7]: yep, this means daily at midnight in UTC timezone
[https://crontab.guru/#00***](https://crontab.guru/#0_0_*_*_*)
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule

**Review by mr-kamran-ali — DISMISSED** [review]
Merge main and good to go.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -3,12 +3,12 @@ default_language_version:
...
 repos:
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: v0.1.13
+    rev: v0.2.0
     hooks:
       - id: ruff
         args: [--fix, --exit-non-zero-on-fix]
   - repo: https://github.com/psf/black
-    rev: 23.12.1
+    rev: 24.1.1
```
[the remark this answers is not in the document]
Comment (cutoffthetop) [reply .pre-commit-config.yaml]: check [db2221a](https://github.com/robert-koch-institut/mex-backend/pull/24/commits/db2221a3e599d86ea202c0656ad07d33f4f907ce)

**Comment by cutoffthetop** [comment]
@mr-kamran-ali main is merged and black updated, could you approve again please?

**Review by mr-kamran-ali — APPROVED** [review]

**Pull request merged**